File: 2129. Capitalize the Title/Solution.js
var capitalizeTitle = function (title) {
  var words = title.split(" ");
  //   console.log(words);

  for (let i = 0; i < words.length; i++) {
    var word = words[i];
    var newword = "";

    for (let j = 0; j < word.length; j++) {
      console.log(word[j]);
      if (j === 0 && word.length > 2) {
        newword += toUpperCase(word[j]);
      } else {
        newword += toLowerCase(word[j]);
      }
    }
    words[i] = newword;
    console.log(words);
  }
  var capitalizedTitle = words.join(" ");
  return capitalizedTitle;
};

function toUpperCase(char) {
  let word = char.charCodeAt(0);
  if (word >= 97 && word <= 122) {
    return String.fromCharCode(word - 32);
  }
  console.log(word);
  return char;
}

function toLowerCase(char) {
  let word = char.charCodeAt(0);
  if (word >= 65 && word <= 90) {
    return String.fromCharCode(word + 32);
  }
  console.log(word);

  return char;
}

console.log(
  capitalizeTitle("thIs Is sohEl FROM RESEARCH ALLIED IN THE FORM OF THE")
);

File: 2129. Capitalize the Title/Solution.java
public class Solution {

    public String capitalizeTitle(String title) {
        String[] words = title.split(" ");
        for (int i = 0; i < words.length; i++) {
            char[] chars = words[i].toCharArray();
            for (int j = 0; j < chars.length; j++) {
                if (j == 0 && chars.length > 2) {
                    chars[j] = Character.toUpperCase(chars[j]);
                } else {
                    chars[j] = Character.toLowerCase(chars[j]);
                }
            }
            words[i] = new String(chars);
        }
        return String.join(" ", words);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String sohel = solution.capitalizeTitle("this is the sohel FROM BHARATNXT");

        System.out.println(sohel);
    }
}
